packages/jazz-tools/src/coValues/deepLoading.ts

```diff
@@ -81,8 +86,8 @@ export function fulfillsDepth(depth: any, value: CoValue): boolean {
                           ).optional,
             );
         }
-    } else if (value._type === "BinaryCoStream") {
-        return true;
+    } else if (isBinaryCoStream(value)) {
+        return value.isBinaryStreamEnded();
```

Comment:
**observation:** change affects not only `loadAsBlob` but also `load` and `ensureLoaded`

I've gone with this road because:
- `loadAsBlob` already uses `load` interally to check if the stream is loaded
- Resolving `load` and `ensureLoaded` when the stream is completed is kinda what I would expect reading at these methods names

The drawback is that when the user deep loads a file from `useCoState` the hook would return a value only when the file is completely downloaded (am I right?)

Example:
```
// me is returned only when all the tracks files are downloaded
const { me } = useAccount({
  root: { activePlaylist: { tracks: [{ file: [] }] } }
});
```

This isn't probably good for the normal use case, where it would be better to get the JSON data as soon as possible and leave the files to be downloaded in background.

Maybe it would be better to change only `loadAsBlob` and leave `load` and `ensureLoaded` as they are.

WDYT @aeplay?

Reply:
Writing this comment made me realize that is better to change only `loadAsBlob`

Opened another PR with a better implementation https://github.com/gardencmp/jazz/pull/337